### PR TITLE
Create RFC-compliant storage manifests

### PR DESCRIPTION
**Pull request by alexwlchan**
So if a bag has a file in a different version, we'll resolve it to create an RFC-compliant path.

Still to do:

- [x] Actually use this in the bag register
- [x] Modify the display models to match

### CODE REVIEW

**Comment by alexwlchan** [comment]
This is deployed and working!

**Comment by alexwlchan** [comment]
Hmm. I want to think about the consistency of version information in the API.

e.g. we use "v2" in path but you request `?version=2`. Should we use "v2" everywhere externally?

File: bag_register/src/test/scala/uk/ac/wellcome/platform/archive/bag_register/fixtures/BagRegisterFixtures.scala [hunk 1]
```diff
@@ -101,4 +102,21 @@ trait BagRegisterFixtures
...
     }
+
+  // The bag register inspects the paths to a bag's entries to
+  // check they are in the correct format post-replicator,
+  // hence the version directory.
+  def withBag[R](bucket: Bucket,
+                 bagInfo: BagInfo,
+                 externalIdentifier: ExternalIdentifier,
+                 space: StorageSpace,
+                 version: Int)(testWith: TestWith[ObjectLocation, R]): R =
```
Comment (kenoir) [inline bag_register/src/test/scala/uk/ac/wellcome/platform/archive/bag_register/fixtures/BagRegisterFixtures.scala]: 👍

File: common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/StorageManifestGenerators.scala [hunk 2]
```diff
@@ -38,21 +37,21 @@ trait StorageManifestGenerators
...
       ),
-      tagManifest = BagManifest(
+      tagManifest = FileManifest(
         checksumAlgorithm,
         files = Seq(
           createBagFile,
           createBagFile,
           createBagFile
-        )
+        ).map { StorageManifestFile.apply }
```
Comment (kenoir) [inline common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/StorageManifestGenerators.scala]: As a style thing I would not make this in-line as I reckon it's easier to read.

Comment (alexwlchan) [reply common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/StorageManifestGenerators.scala]: oh hold on, that shouldn't be there

Comment (alexwlchan) [reply common/src/test/scala/uk/ac/wellcome/platform/archive/common/generators/StorageManifestGenerators.scala]: This was temporary code to keep that generator compiling while I was fiddling elsewhere, but I meant to take it out

**Pull request merged**